server/controllers/gradeController.js

```diff
@@ -129,20 +120,27 @@ async function createFixture(req, res, next) {
     try {
         const { teams, grade, body: { numRounds, datesAndLocations } } = req
 
+        for (const team of teams) {
+            if (!await _addTeamToGrade(team, grade, req.season, next)) return
```

Comment:
This is kinda dodgy code but I was running into an issue where the for loop would continue even when I called the `next` middleware in `_addTeamToGrade`

Reply:
i think this will still have some weird behaviour with partially adding some of the teams but not the rest if there is an error midway through, but should be okay